pandas/core/indexes/multi.py

```diff
@@ -3095,8 +3106,56 @@ def _update_indexer(idxr, indexer=indexer):
         # empty indexer
         if indexer is None:
             return Int64Index([])._ndarray_values
+
+        if need_sort:
+            indexer = self._reorder_indexer(seq, indexer)
+
         return indexer._ndarray_values
 
+    def _reorder_indexer(
+        self, seq: Tuple[Union[Scalar, Iterable, AnyArrayLike], ...], indexer: ArrayLike
+    ) -> ArrayLike:
+        """
+        Reorder an indexer of a MultiIndex (self) so that the label are in the
+        same order as given in seq
+
+        Parameters
+        ----------
+        seq : label/slice/list/mask or a sequence of such
+        indexer: an Int64Index indexer of self
+
+        Returns
+        -------
+        indexer : a sorted Int64Index indexer of self ordered as seq
+        """
+        n = len(self)
+        keys = tuple()  # type: Tuple[np.ndarray, ...]
```

Comment:
can you use explict typing (we dropped 3.5)

Reply:
Done